public/app/features/profile/UserProfileEditTabs.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { type ComponentType, Fragment, type ReactElement, useCallback, useMemo } from 'react';
+
+import { type UrlQueryValue } from '@grafana/data';
+import { selectors } from '@grafana/e2e-selectors';
+import { Stack, Tab, TabContent, TabsBar } from '@grafana/ui';
+import { useQueryParams } from 'app/core/hooks/useQueryParams';
+import { t } from 'app/core/internationalization';
+
+const TAB_QUERY_PARAM = 'tab';
+const GENERAL_SETTINGS_TAB = 'general';
+
+type Props = {
+  children?: React.ReactNode;
+  components: Array<ComponentType & { meta: { title: string } }>;
```

Comment:
Couldn't we use `ComponentTypeWithMeta` here as well?

```suggestion
  components: ComponentTypeWithMeta[];
```